### PR TITLE
Fix signature of asyncio.create_connection.

### PR DESCRIPTION
**Pull request by aaugustin**
* local_addr and server_hostname are optional and default to None.
* If sock is given, none of host, port, family, proto, flags and
  local_addr should be specified.

### CODE REVIEW

File: stdlib/3/asyncio/events.pyi [hunk 1]
```diff
@@ -103,11 +103,18 @@ class AbstractEventLoop(metaclass=ABCMeta):
...
     def create_connection(self, protocol_factory: _ProtocolFactory, host: str = ..., port: int = ..., *,
-                          ssl: _SSLContext = ..., family: int = ..., proto: int = ..., flags: int = ..., sock: Optional[socket] = ...,
-                          local_addr: str = ..., server_hostname: str = ...) -> Generator[Any, None, _TransProtPair]: ...
+                          ssl: _SSLContext = ..., family: int = ..., proto: int = ..., flags: int = ..., sock: None = ...,
+                          local_addr: Optional[str] = ..., server_hostname: Optional[str] = ...) -> Generator[Any, None, _TransProtPair]: ...
+    @overload
+    @abstractmethod
+    @coroutine
+    def create_connection(self, protocol_factory: _ProtocolFactory, host: None = ..., port: None = ..., *,
+                          ssl: _SSLContext = ..., family: int = ..., proto: int = ..., flags: int = ..., sock: socket = ...,
```
Comment (JelleZijlstra) [inline stdlib/3/asyncio/events.pyi]: Shouldn't we remove the default value for `sock` in this case?

Comment (aaugustin) [reply stdlib/3/asyncio/events.pyi]: I'm sorry — I don't understand what you mean by "remove the default value". In `sock: socket = ...` the default value is `Ellipsis` so I'd say it's already removed. I may be missing something obvious.

Comment (JelleZijlstra) [reply stdlib/3/asyncio/events.pyi]: I mean just writing `sock: socket`, without the `= ...`. That signifies that for this overload, the `sock` argument must be passed.

Comment (aaugustin) [reply stdlib/3/asyncio/events.pyi]: OK. Thanks for the explanation, I didn't know that!

I made that change — and also rebased on top of master to fix conflicts.

Comment (aaugustin) [reply stdlib/3/asyncio/events.pyi]: Also updated `create_server` for consistency. I had written it in the same style in #2763.

**Comment by aaugustin** [comment]
Thanks!

**Pull request merged**